### PR TITLE
Further clarify requirements for PUT-to-create and PATCH-to-create

### PR DESCRIPTION
**Pull request by michielbdejong**
ACL permission requirements are hard to follow even if you spell them out completely, and in this spec text we try to speak in generic terms about them, and only mention the aspects that are worth noting, and that makes it even harder.

I might be getting this wrong because it's all so complex, but here's what I think now after reading the recent comments:

It seems #122 created [some confusion](https://github.com/solid-contrib/web-access-control-tests/pull/56). 

This updates the text to match what we decided [in response to a similar but unrelated confusion](https://github.com/solid/web-access-control-spec/issues/105#issuecomment-1160876522).

At the time we decided not to update the spec text, but now that the spec text is more detailed, the current statement is not correctly conveying that access to both the containing folder and the non-existing resource URL is required.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -805,10 +805,10 @@ <h5 property="schema:name"><span>Note</span>: HTTP Method and Access Mode Mappin
...
 
                             <dt id="http-patch"><code>PATCH</code></dt>
-                            <dd>As the processing of HTTP <code>PATCH</code> method requests depends on the request semantics and content, <code>acl:Append</code> can allow requests using SPARQL 1.1 Update’s [<cite><a class="bibref" href="#bib-sparql11-update">SPARQL11-UPDATE</a></cite>] <code>INSERT DATA</code> operation but not <code>DELETE DATA</code>, whereas <code>acl:Write</code> would allow both operations.</dd>
+                            <dd>As the processing of HTTP <code>PATCH</code> method requests depends on the request semantics and content, <code>acl:Append</code> can allow requests using SPARQL 1.1 Update’s [<cite><a class="bibref" href="#bib-sparql11-update">SPARQL11-UPDATE</a></cite>] <code>INSERT DATA</code> operation but not <code>DELETE DATA</code>, whereas <code>acl:Write</code> would allow both operations. And again, to create a new resource, access to the container additionally needs to be either <code>acl:Append</code> or <code>acl:Write</code>.</dd>
```
Comment (TallTed) [inline index.html:811]:
```suggestion
                            <dd>As the processing of HTTP <code>PATCH</code> method requests depends on the request semantics and content, <code>acl:Append</code> can allow requests using SPARQL 1.1 Update’s [<cite><a class="bibref" href="#bib-sparql11-update">SPARQL11-UPDATE</a></cite>] <code>INSERT DATA</code> operation but not <code>DELETE DATA</code>, whereas <code>acl:Write</code> would allow both operations. As with <code>PUT</code>, to <i>create a new</i> resource, the requester needs <code>acl:Append</code> and/or <code>acl:Write</code> access to the container as well as <code>acl:Write</code> access to the (new) resource.</dd>
```

File: index.html [hunk 2]
```diff
@@ -805,10 +805,10 @@ <h5 property="schema:name"><span>Note</span>: HTTP Method and Access Mode Mappin
                             <dd>The HTTP <code>POST</code> can be used to create a new resource in a container or add information to existing resources (but not remove resources or its contents) with either <code>acl:Append</code> or <code>acl:Write</code>.</dd>
 
                             <dt id="http-put"><code>PUT</code></dt>
-                            <dd>As the HTTP <code>PUT</code> method requests to create or replace the resource state, the <code>acl:Write</code> access mode is required to replace an existing resource, but to create a new resource, access to the container can be either <code>acl:Append</code> or <code>acl:Write</code>.</dd>
+                            <dd>As the HTTP <code>PUT</code> method requests to create or replace the resource state, the <code>acl:Write</code> access mode is required to replace an existing resource, but to create a new resource, access to the container additionally needs to be either <code>acl:Append</code> or <code>acl:Write</code>.</dd>
```
Comment (csarven) [inline index.html:808]:
```suggestion
                            <dd>The HTTP <code>PUT</code> method requests to create or replace the resource state. To create a new resource, <code>acl:Append</code> or <code>acl:Write</code> access mode to the container as well as <code>acl:Write</code> access mode to the new resource is required. To replace an existing resource,  only <code>acl:Write</code> access mode to the resource itself is required.</dd>
```

[the remark this answers is not in the document]
Comment (csarven) [reply index.html:808]: * "the requester" needing particular access is not necessarily true or at least is subject to misinterpretation. I'd suggest to instead keep focus on the modes as this non-normative section is specifically about mapping the HTTP request semantics to access modes.
* "and/or" can be simplified to "or" as the the latter inherently covers both being true. So, I suggest to keep the original text
* `<i>` is not needed. No objection to leaving it in.

File: index.html (hunk 1 above)
Comment (csarven) [inline index.html:811]: Combines / edits feedback from Michiel and Ted, as well as Jeff (in chats somewhere) and https://github.com/solid/web-access-control-spec/issues/104#issuecomment-1146838683 (removing mention of SPARQL Update as that's not required but was only used as an example):

```suggestion
                            <dd>As the processing of HTTP <code>PATCH</code> method requests depends on the request semantics and content, <code>acl:Append</code> can allow requests to <em>insert</em> but not <em>delete<em> operations, whereas <code>acl:Write</code> would allow both operations. To create a new resource, <code>acl:Append</code> or <code>acl:Write</code> access mode to the container is additionally required.</dd>
```

Comment (csarven) [reply index.html:811]: My feedback re "requester" and `<i>` in other comment applies here as well. I think the original suggestion by Michiel is simpler to parse, and I think these definitions should stand on their own - without for example needing to reference PUT. One reason for that is that if the PUT request semantics with access mode mapping changes in WAC, PATCH's doesn't necessarily need to change. They are independent. This is also true when WAC is used alongside a resource Protocol (Solid Protocol, LDP, AP..)

**Review by csarven — CHANGES_REQUESTED** [review]
@michielbdejong , the requested clarification is good / in the right direction, thanks!

I've requested changes building on suggestions made by @michielbdejong @TallTed .

File: index.html (hunk 1 above)
Comment (TallTed) [reply index.html:811]: I can live with it.

Comment (TallTed) [reply index.html:811]: I can live with it.

File: index.html (hunk 2 above)
[the remark this answers is not in the document]
Comment (TallTed) [reply index.html:808]: * I've removed `the requester`.
* `"and/or" can be simplified to "or"` — Technically true. However, many people read `or` as `xor` rather than as the intended `and/or`. This is why I try to always write `and/or` when that is the intended meaning.
* `<i>` is not *needed*. To me, the extra emphasis on the desired action helps to make the separation of required access mode(s) clearer. I will live without it if need be.
```suggestion
                            <dd>The HTTP <code>PUT</code> method requests to create or replace the resource state. <i>Creating</i> a new resource requires <code>acl:Append</code> and/or <code>acl:Write</code> access to the container as well as <code>acl:Write</code> access to the (new) resource. <i>Replacing</i> an existing resource requires only <code>acl:Write</code> access to the resource itself.</dd>
```

Comment (TallTed) [reply index.html:808]: See https://github.com/solid/web-access-control-spec/pull/128/files#r1425884743

**Review by csarven — APPROVED** [review]
